daliuge-engine/dlg/io.py

```diff
@@ -426,14 +429,20 @@ def __init__(self, object_id, plasma_path='/tmp/plasma'):
         self._object_id = object_id
 
     def _open(self, **kwargs):
+        self._done = False
         return plasma.connect(self._plasma_path)
 
     def _close(self, **kwargs):
         pass
 
     def _read(self, count, **kwargs):
-        data = self._desc.get(self._object_id)
-        return data
+        if not self._done:
+            # basic API will encapsulate the entire transfer
+            data = self._desc.get(self._object_id)
+            self._done = True
+            return data
+        else:
+            return []
```

Comment:
This seemed to have been a bug before as I found it would read the store contents over and over in a loop